play-ebean/src/main/java/play/db/ebean/EbeanDynamicEvolutions.java

```diff
@@ -65,9 +66,22 @@ public void create() {
                 String evolutionScript = generateEvolutionScript(servers.get(key), serverConfig);
                 if (evolutionScript != null) {
                     File evolutions = environment.getFile("conf/evolutions/" + key + "/1.sql");
-                    if (!evolutions.exists() || Files.readFile(evolutions).startsWith("# --- Created by Ebean DDL")) {
-                        Files.createDirectory(environment.getFile("conf/evolutions/" + key));
-                        Files.writeFileIfChanged(evolutions, evolutionScript);
+                    try {
+                        String content;
+                        if (!evolutions.exists()) {
+                            content = "";
+                        } else {
+                            content = new String(Files.readAllBytes(evolutions.toPath()), "utf-8");
+                        }
+
+                        if (content.isEmpty() || content.startsWith("# --- Created by Ebean DDL")) {
+                            environment.getFile("conf/evolutions/" + key).mkdirs();
+                            if (!content.equals(evolutionScript)) {
+                                Files.write(evolutions.toPath(), evolutionScript.getBytes("utf-8"));
+                            }
+                        }
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
```

Comment:
UsefulException should be thrown if you have useful information to attach to it.  I can't think of any reason why these IO exceptions would be thrown, it's just that because it's Java I'm forced to handle them (previously, they were thrown from the Scala code, so they didn't need to be checked).  So there's no useful information to attach here.